lib/endpoints/boot_params.js

```diff
@@ -5,16 +5,70 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright 2020 Joyent, Inc.
  */
+var util = require('util');
 
 var restify = require('restify');
 
 var ModelServer = require('../models/server');
 var common = require('../common');
 var validation = require('../validation/endpoints');
 var ModelPlatform = require('../models/platform');
-var dns = require('dns');
+
+/*
+ * Expected boot modules format is as follows:
+ *
+ * {
+ *      "path": "boot module path to be used by booter",
+ *      "type": "Content type. Right now only base64 is supported",
+ *      "content": "File contents, if necessary encoded as specified by type"
+ * }
+ *
+ * In the future, support for new content types or additionaly members for
+ * the boot_module objects could be provided, being the defaults the originally
+ * supported values.
+ */
+function validateBootModules(boot_modules) {
+    var errs = [];
+    var i;
+    for (i = 0; i < boot_modules.length; i += 1) {
+        var bm = boot_modules[i];
+        if (!bm.path || !bm.content) {
```

Comment:
the "content" check should logically go after the part where we check bm.type I think?

Reply:
Sounds good.

Reply:
This wasn't moved?

Reply:
Sorry, I understood we should do the type format verification after we check the content was of the required type.

My point here is: shall we care about content type at all if there is no content or no path is specified? In case those are present, then we care about type being set to something unexpected and, only then, we verify that the provided content matches the provided type. Has sense?

Reply:
Imagine that we add a new "type", but somebody tries to use it against this code without it being updated. That will then give the wrong error message - it should be complaining that the type is unsupported.

So logically we should be checking type first, and reporting errors on that. Then, if we see base64 as the type, we should be looking for path and content.

Reply:
I don't think we're gonna add any new type w/o checking this but ... who knows! Changed as you suggested.